wagtail/rich_text/__init__.py

```diff
@@ -132,7 +132,8 @@ class EmbedHandler(EntityHandler):
 class RichTextMaxLengthValidator(MaxLengthValidator):
     """
     A variant of MaxLengthValidator that only counts text (not HTML tags) towards the limit
+    Un-escapes entities for consistency with client-side character count.
     """
 
     def clean(self, x):
-        return len(strip_tags(x))
+        return len(unescape(strip_tags(x)))
```

Comment:
This is similar to `get_text_for_indexing` further up in this file, though for the length we completely disregard line breaks / the fact content is over multiple blocks.